### PR TITLE
ci: update tests workflow to use centralized reusable workflow and auto-suggest formatting changes on PRs

### PR DESCRIPTION
**Pull request by thazhemadam**
Follow up to #183.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@ErikQQY is it sometimes an array sometimes an AbstractVectorOfArray? We should probably always make it an AbstractVectorofArray

**Comment by ErikQQY** [comment]
> @ErikQQY is it sometimes an array sometimes an AbstractVectorOfArray? We should probably always make it an AbstractVectorofArray

We always use `Vector{Vector}}` type in the nonlinear problems constructing and solving, should we change all of them to the `AbstractVectorOfArray` type instead? That could make the recursive flatten and unflatten of a `Vector{Vector}` much easier.

**Comment by ChrisRackauckas** [comment]
Yes, Make it an AbstractDiffEqArray, since there's time associated with it as well.

**Comment by ChrisRackauckas** [comment]
@ErikQQY can you help with the completion of this?

**Comment by ErikQQY** [comment]
Sure, I will help to get this complete

**Comment by ErikQQY** [comment]
The errors in tthis PR are mostly caused by the deprecation of indexing methods in RecursiveArrayTools.jl:

There are two ways of specifying boundary conditions when constructing a `BVProblem`, e.g. `bc(u,p,t)=[u[1][1]]` or `bc(u,p,t)=[u(10.0)[1]]` before RecursiveArrayTools.jl deprecate `u[1]` for `u.u[1]` or `u[:,1]` kind indexing, both of them were working fine, but now the centralized CI errors are mostly caused by the deprecation of this API.

So here is the solution:

We need to change the convention of defining boundary conditions as `bc(u,p,t)=[u.u[1][1]]` or `bc(u,p,t)=[u[:,1][1]]` now for this PR to continue. **This may involve lots of tests, examples, and docs change.**

**Comment by ChrisRackauckas** [comment]
> We need to change the convention of defining boundary conditions as bc(u,p,t)=[u.u[1][1]] or bc(u,p,t)=[u[:,1][1]] now for this PR to continue. This may involve lots of tests, examples, and docs change.

Yes, that's what needs to happen everywhere.

File: src/default_nlsolve.jl [hunk 1]
```diff
@@ -18,7 +18,7 @@ function __FastShortcutBVPCompatibleNLLSPolyalg(
                 linsolve, precs, disable_geodesic = Val(true), autodiff, kwargs...),
             TrustRegion(; concrete_jac, linsolve, precs, autodiff, kwargs...),
             GaussNewton(; concrete_jac, linsolve, precs,
-                linesearch = LineSearchesJL(; method = BackTracking()),
+                linesearch = NonlinearSolve.LineSearchesJL(; method = BackTracking()),
```
Comment (ChrisRackauckas) [inline src/default_nlsolve.jl:21]:
```suggestion
                linesearch = BackTracking(),
```

File: src/default_nlsolve.jl [hunk 2]
```diff
@@ -33,7 +33,8 @@ function __FastShortcutBVPCompatibleNonlinearPolyalg(
     else
         algs = (NewtonRaphson(; concrete_jac, linsolve, precs, autodiff),
             NewtonRaphson(; concrete_jac, linsolve, precs,
-                linesearch = LineSearchesJL(; method = BackTracking()), autodiff),
+                linesearch = NonlinearSolve.LineSearchesJL(; method = BackTracking()),
```
Comment (ChrisRackauckas) [inline src/default_nlsolve.jl:36]:
```suggestion
                linesearch = BackTracking(),
```

**Comment by ChrisRackauckas** [comment]
@avik-pal do you know why this ambiguity is still reported? I don't see it actually being hit?

**Comment by avik-pal** [comment]
https://github.com/SciML/BoundaryValueDiffEq.jl/blob/master/test/misc/qa_tests.jl#L5 needs to be removed

**Comment by ErikQQY** [comment]
The Windows CI test looks weird, all other tests are working fine
